test/integration/replicaset/test_replicaset_upgrade.py

```diff
@@ -192,3 +194,53 @@ def test_upgrade_downgraded_cluster_replicasets(tt_cmd, tmp_path):
 
     finally:
         app.stop()
+
+
+def start_application(cmd, workdir, app_name, instances):
+    instance_process = subprocess.Popen(
+        cmd,
+        cwd=workdir,
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE,
+        text=True
+    )
+    start_output = instance_process.stdout.read()
+    for inst in instances:
+        assert f"Starting an instance [{app_name}:{inst}]" in start_output
+
+
+@pytest.mark.skipif(tarantool_major_version < 3,
+                    reason="skip cluster instances test for Tarantool < 3")
+def test_upgrade_remote_replicasets(tt_cmd, tmpdir_with_cfg):
+    tmpdir = tmpdir_with_cfg
+    app_name = "small_cluster_app"
+    app_path = os.path.join(tmpdir, app_name)
+    shutil.copytree(os.path.join(os.path.dirname(__file__), app_name), app_path)
+
+    run_dir = os.path.join(tmpdir, app_name, run_path)
+    instances = ['storage-master', 'storage-replica']
+
+    try:
+        # Start an instance.
+        start_cmd = [tt_cmd, "start", app_name]
+        start_application(start_cmd, tmpdir, app_name, instances)
+
+        # Check status.
+        for inst in instances:
+            file = wait_file(os.path.join(run_dir, inst), 'tarantool.pid', [])
+            assert file != ""
+            file = wait_file(os.path.join(run_dir, inst), control_socket, [])
+            assert file != ""
+        # App is ready, but It is necessary to wait for some time
+        # to be able to establish a connection.
+        time.sleep(3)
```

Comment:
Is there any way to do this without using `sleep`? I mean this potentially creates a flacking test.

Reply:
You could try the approach: https://github.com/tarantool/tt/blob/7d8a6cfc027e2b9403c1a77e0b35c32cf25feaed/test/integration/replicaset/cluster_app_failovers/init.lua#L18-L19

Create a file/open a port from a server when you are sure that it is already bootstrapped.

Reply:
Thank you! It really worked.